packages/scxa-experiment-page/src/TSnePlotViewRoute.js

```diff
@@ -0,0 +1,338 @@
+                this.setState({selectedClusterId : colourByValue})
+                queryParams.push({k: colourByValue})
+              } else {
+                this.setState({selectedColourBy : colourByValue,})
+                queryParams.push({colourBy: colourByValue})
+              }
+
+              updateUrlWithParams(queryParams)
+            }
+          }
+        />
+      },
+      {
+        path: `/marker-genes`,
+        title: `Marker Genes`,
+        main: () => <ClustersHeatmapView
+          host={atlasUrl}
+          resource={
+            this.state.selectedColourByCategory === METADATA_PLOT ?
+                URI(`json/experiments/${this.state.experimentAccession}/marker-genes-heatmap/cell-types`)
+                    .search({cellGroupType: this.state.selectedColourBy})
+                    .toString() :
+                URI(`json/experiments/${this.state.experimentAccession}/marker-genes/clusters`)
+                    .search({k: this.state.selectedClusterId})
+                    .toString()
+          }
+          wrapperClassName={`row expanded`}
+          ks={ks}
+          selectedClusterByCategory={search.cellGroupType || search.k || preferredK}
+          selectedK={this.state.selectedColourByCategory === METADATA_PLOT ? this.state.selectedColourBy : this.state.selectedClusterId}
+          onSelectK={(colourByValue) => {
+            // If marker gene heatmap is coloured by numeric k values
+            let isMetadataCluster = parseInt(colourByValue) ? false : true
+            if(isMetadataCluster) {
+              this.setState({
+                selectedColourBy : colourByValue,
+                selectedColourByCategory : METADATA_PLOT
+              })
+              updateUrlWithParams( [{colourBy: colourByValue}])
+            } else {
+              this.setState({
+                selectedClusterId : colourByValue,
+                selectedColourByCategory : CLUSTERS_PLOT
+              })
+              updateUrlWithParams([{k: colourByValue}])
+            }
+           }
+          }
+          onChangeMarkerGeneFor={(selectedOption) => {
+            this.setState((state) => ({
+              selectedClusterIdOption: selectedOption
+            }))
+          }}
+          ksWithMarkers={ksWithMarkerGenes}
+          metadata={metadata.map(data => {return {value: data.value.replaceAll(`_`, ` `), label: data.label}})}
+          species={species}
+          heatmapType={this.state.selectedColourByCategory === METADATA_PLOT ? CELL_TYPE_MARKER_GENE_HEATMAP : CLUSTER_MARKER_GENE_HEATMAP}
+        />
+      },
+      {
+        path: `/anatomogram`,
+        title: `Anatomogram`,
+        main: () =>
+          <AnatomogramCellTypeHeatmapView
+            showIds={anatomogram[organWithMostOntologies]}
+            experimentAccession={experimentAccession}
+            species={species}
+            organ={organWithMostOntologies}
+            host={atlasUrl}
+          />
+      },
+      {
```

Comment:
Is it possible to extract this routes element (currently 4) to their own files.
That might make this file a bit cleaner.